### ISSUE TITLE
 Add support for color changing escape sequences

### ISSUE THREAD

**Issue by dylanaraps** [vercel/hyper#2038]
- [x] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
- [x] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate
- **OS version and name**: Arch Linux
- **Hyper.app version**: `1.3.3`
- **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: No changes made to default config. (N/A)
- **Relevant information from devtools**: N/A
- **The issue is reproducible in vanilla Hyper.app**: Yes

## Issue

These escape sequences manipulate the open terminal's colorscheme and allow you to change your terminal's colorscheme on the fly. These escape sequences are supported by all of the terminal emulators I've tried with the exception of `konsole` which ignores them.

Two of my projects use these escape sequences for their main purpose and they currently don't work in Alacritty since the sequences aren't yet supported. 

- https://github.com/dylanaraps/pywal
- https://github.com/dylanaraps/wal

I fully understand if you don't want to implement this feature, there are a lot of really weird escape sequences in the Xterm/Rxvt specs and I understand that you want to keep Hyper simple. \^\^

The sequences are as follows:

```
# Manipulate colors 0-256
# \033]4;{index};{color}\007

# Change color 7 to #FFFFFF
\033]4;7;#FFFFFF\007

# Change color 14 to #333333
\033]4;14;#333333\007

# Manipulate special colors.
# 10 = foreground, 11 = background, 12 = cursor foregound
# 13 = mouse foreground, 708 = terminal border background
# \033]{index};{color}\007

# Change the terminal foreground to #FFFFFF
\033]10;#FFFFFF\007

# Change the terminal background to #000000
\033]11;#000000\007

# Change the terminal cursor to #FFFFFF
\033]12;#FFFFFF\007

# Change the terminal border background to #000000
\033]708;#000000\007
```

Source:

- http://pod.tst.eu/http://cvs.schmorp.de/rxvt-unicode/doc/rxvt.7.pod#XTerm_Operating_System_Commands

More information:

- https://github.com/jwilm/alacritty/issues/656
- https://github.com/jwilm/alacritty/pull/662

**Comment by Stanzilla** [comment]
Hey there, to verify if this is indeed a Hyper issue or upstream at https://github.com/xtermjs/xterm.js/ could you please test if another xterm.js powered terminal shows the same problem you are experiencing? I suggest using https://github.com/Microsoft/vscode for testing. Thanks!

**Comment by johnsonwj** [comment]
PS - in case this wasn't obvious from the screenshots, I am on Windows 10 ... I don't have an arch box handy to try to repro this from the original report :(

**Comment by johnsonwj** [comment]
This problem does not appear on OSX.

![image](https://user-images.githubusercontent.com/3478488/39321936-30428d30-493d-11e8-88ec-bf53cee8db2a.png)

**Comment by dylanaraps** [comment]
I’ll test this on Arch in an hour or so.

**Comment by dylanaraps** [comment]
The sequences do not work in `vscode`.

**Comment by johnsonwj** [comment]
Yeah, actually today they are not working for me in VS Code either...I am not sure what changed in the meantime. I am using the same version.

**Comment by vivekvjn** [comment]
Try `echo -e '\e[31m   767M \e[41m____________________\e[m \e[31mdir1/\e[m'`
It works for me in putty, VS Code etc. 
![image](https://user-images.githubusercontent.com/6826760/46421953-331f2380-c72b-11e8-8363-290ac2da85ac.png)
but not on Hyper. 
![image](https://user-images.githubusercontent.com/6826760/46422013-52b64c00-c72b-11e8-9a85-fc052f8245cd.png)

Using Hyper 2.0.0 on WSL
![image](https://user-images.githubusercontent.com/6826760/46422105-85f8db00-c72b-11e8-8f94-714220b164d5.png)

**Comment by seve** [comment]
Wondering about the status of this issue.  Is there a current fix for changing the colorscheme via wal/pywal?

**Comment by matteomeli** [comment]
I got around this issue adding a value in the registry. Add a **REG_DWORD** type variable called **VirtualTerminalLevel** with an hexadecimal value of 1 at **Computer\HKEY_CURRENT_USER\Console** and restart hyper. It should recognize color escape sequences correctly then.

**Comment by LabhanshAgrawal** [comment]
Tested `echo -e '\e[31m 767M \e[41m____________________\e[m \e[31mdir1/\e[m'`
on mac, windows & ubuntu using the latest [canary build](https://github.com/vercel/hyper/actions/runs/153185087)
It seems to be working well
Feel free to open a new issue if you face any other problems

**Comment by dylanaraps** [comment]
@LabhanshAgrawal Those aren't the sequences I am talking about. Please reopen this issue and please carefully read my initial post. :+1:

**Comment by LabhanshAgrawal** [comment]
I had closed this based on earlier comment by vivek
@dylanaraps can you give any command to test it
I get this when I run the command from pywal wiki
![Screenshot 2021-02-08 at 12 26 50](https://user-images.githubusercontent.com/16598275/107185758-066d2c00-6a09-11eb-9a54-3a0984601b39.png)
nothing is printed or changed in the rest of the terminal for me
is this ok?

**Comment by ihucos** [comment]
This is a command to test it. When running this on e.G. xterm, the color scheme will change to solarized dark.

(As always take a look at it before pasting into the terminal)

```
printf '\x1b]12;#dc302f\x07\x1b]11;#002b36\x07\x1b]10;#839496\x07\x1b]708;#002b36\x07\x1b]4;0;#073642\x07\x1b]4;1;#dc302f\x07\x1b]4;2;#859900\x07\x1b]4;3;#b58900\x07\x1b]4;4;#268bd2\x07\x1b]4;5;#d33681\x07\x1b]4;6;#29a198\x07\x1b]4;7;#eee8d5\x07\x1b]4;9;#cb4b16\x07\x1b]4;8;#002b36\x07\x1b]4;10;#586d75\x07\x1b]4;11;#657b83\x07\x1b]4;12;#839496\x07\x1b]4;13;#6c71c4\x07\x1b]4;14;#93a1a1\x07\x1b]4;15;#fdf6e3\x07'
```

**Comment by rstacruz** [comment]
I'm confirming that this works well on Windows 11 with Hyper 3.1.5.

![image](https://user-images.githubusercontent.com/74385/150041294-7214eef2-cb5e-4ad4-a251-6abae0e9302f.png)

**Comment by LabhanshAgrawal** [comment]
Closing as fixed (added in xterm 4.16)
